### ISSUE TITLE
Asserting sent request fails on first test when recording responses

### ISSUE THREAD

**Issue by mdpoulter** [Sammyjo20/Saloon#322]
Thanks so much for such a great package - it's amazing seeing the effort you put into it!

I've encountered a problem with fixtures which feels like a bug, but may also be intentional. I have created a minimal example of the test I'm trying to run:

```php
use Package\SomeConnector;
use Package\SomeRequest;
use Saloon\Http\Faking\MockResponse;
use Saloon\Http\PendingRequest;
use Saloon\Laravel\Facades\Saloon;

it('asserts request was sent', function (): void {
    Saloon::fake([
        '*' => fn (PendingRequest $pendingRequest) => MockResponse::fixture(/** name logic  */),
    ]);

    (new SomeConnector())->send(new SomeRequest());

    Saloon::assertSent(SomeRequest::class);
});
```

What I would expect is that this test passes every time, even the very first time when it is recording the response and creating the fixture. Instead, the assertion fails on the first run. The fixture gets created though, and the test then passes on all subsequent runs.

It may be useful to include that I'm using a connector and request from an external SDK package I created.

Is this expected behaviour?

**Comment by Sammyjo20** [comment]
Hey @mdpoulter 

Thank you for your kind words and thank you for opening this issue!

This was actually something I was aware of but I wasn't sure if it was just me. I've had this issue when the request I've been recording was non-JSON - what was the request type you were recording? Was it just a JSON response or was it something else like a file or XML?

**Comment by mdpoulter** [comment]
They're all JSON responses actually, and it's happening for GET, POST and DELETE requests (that I've tested). I'm setting the Accept header using the built-in `AcceptsJson` trait too and the saved fixture is definitely JSON.

Doing a little bit of debugging, it seems to be failing because `checkHistoryEmpty()` is returning `true`.

**Comment by Sammyjo20** [comment]
Thanks for the info. I'll do a bit of digging. Does it still fail if you remove the `assertSent` at the bottom there?

**Comment by mdpoulter** [comment]
No, it is the `assertSent` which fails, so without it the test runs fine (although with no assertions then). The fixture gets created with or without the `assertSent`.

This is the exact message from the first test run with the `assertSent`:

```
An expected request was not sent.
Failed asserting that false is true.
at vendor/saloonphp/saloon/src/Http/Faking/MockClient.php:248
at vendor/saloonphp/laravel-plugin/src/Saloon.php:55
at vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:353
```

**Comment by Sammyjo20** [comment]
Oh I see! Okay I will take a further look.

**Comment by Sammyjo20** [comment]
Hey @mdpoulter this should now be fixed in v3.0.4 - I am just tagging a release to fix this in v2 too.
